### PR TITLE
Validation apis, events and version

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:
Add validation to:
- APISpecDefinitionInput
- APISpecInput
- EventAPISpecDefinitionInput
- EventAPISpecInput
- VersionInput

**Related issue(s)**

#255

### CODE REVIEW

File: components/director/examples/add-api/add-api.graphql [hunk 1]
```diff
@@ -4,7 +4,7 @@ mutation {
     applicationID: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
     in: {
       name: "new-api-name"
-      targetURL: "new-api-url"
```
Comment (dbadura) [inline components/director/examples/add-api/add-api.graphql]: Here can be a problem after switching to new our validator.

File: tests/end-to-end/director/validation_test.go [hunk 2]
```diff
@@ -237,6 +237,92 @@ func TestUpdateIntegrationSystem_Validation(t *testing.T) {
...
+	ctx := context.TODO()
+	app := createApplication(t, ctx, "name")
+	defer deleteApplication(t, app.ID)
+
+	api := graphql.APIDefinitionInput{Name: "name", TargetURL: "https://kyma project.io"}
+	apiGQL, err := tc.graphqlizer.APIDefinitionInputToGQL(api)
+	require.NoError(t, err)
+	addAPIRequest := fixAddAPIRequest(app.ID, apiGQL)
+	//WHEN
+	err = tc.RunOperation(ctx, addAPIRequest, nil)
```
Comment (dbadura) [inline tests/end-to-end/director/validation_test.go]: add new lines before all `//`

File: tests/end-to-end/e2e/e2e_test.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func TestCompassAuth(t *testing.T) {
 	t.Log("Add API Spec to Application")
 	apiInput := graphql.APIDefinitionInput{
 		Name:      "new-api-name",
-		TargetURL: "new-api-url",
+		TargetURL: "https://kymaproject.io",
```
Comment (dbadura) [inline tests/end-to-end/e2e/e2e_test.go]: `https://kyma-project.io`

File: components/director/pkg/graphql/version_validation_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+			Value: nil,
+			Valid: true,
+		},
+		{
+			Name:  "Empty string",
+			Value: str.Ptr(inputvalidationtest.EmptyString),
+			Valid: true,
+		},
+		{
+			Name:  "String longer than 257 chars",
```
Comment (kfurgol) [inline components/director/pkg/graphql/version_validation_test.go]:
```suggestion
			Name:  "String longer than 256 chars",
```

File: components/director/pkg/graphql/api_validation_test.go [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			//GIVEN
+			obj := fixValidAPIDefinitionInput()
+			obj.Name = testCase.Value
+			//WHEN
+			err := obj.Validate()
+			//THEN
+			if testCase.Valid {
```
Comment (crabtree) [inline components/director/pkg/graphql/api_validation_test.go]:
```suggestion
			if testCase.ExpectValid {
```

File: components/director/pkg/graphql/event_api_validation.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		validation.Field(&i.Group, validation.RuneLength(0, groupLengthLimit)),
+		validation.Field(&i.Version, validation.NilOrNotEmpty),
+	)
+}
+
+func (i EventAPISpecInput) Validate() error {
+	return validation.Errors{
+		"Rule.Type":                  validation.Validate(&i.EventSpecType, validation.Required, validation.In(EventAPISpecTypeAsyncAPI)),
+		"Rule.Format":                validation.Validate(&i.Format, validation.Required, validation.In(SpecFormatYaml, SpecFormatJSON)),
+		"Rule.MatchingTypeAndFormat": i.validateMatchingSpecAndType(),
```
Comment (crabtree) [inline components/director/pkg/graphql/event_api_validation.go]:
```suggestion
		"Rule.MatchingTypeAndFormat": i.validateTypeWithMatchingSpecFormat(),
```

File: components/director/pkg/graphql/event_api_validation.go [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		"Rule.FetchRequest":          validation.Validate(&i.FetchRequest),
+		"Rule.DataOrFetchRequest":    inputvalidation.ValidateExactlyOneNotNil("Only one of Data or Fetch Request must be passed", i.Data, i.FetchRequest),
+	}.Filter()
+}
+
+func (i EventAPISpecInput) validateMatchingSpecAndType() error {
```
Comment (crabtree) [inline components/director/pkg/graphql/event_api_validation.go]:
```suggestion
func (i EventAPISpecInput) validateTypeWithMatchingSpecFormat() error {
```

File: components/director/pkg/graphql/event_api_validation.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+			return errors.Errorf("format %s is not a valid spec format for spec type %s", i.Format, i.EventSpecType)
+		}
+	default:
+		return errors.New("invalid spec type")
+	}
+	return nil
+}
+
+func (i *EventAPISpecInput) formatIsOneOf(formats []SpecFormat) bool {
+	var slice []string
```
Comment (crabtree) [inline components/director/pkg/graphql/event_api_validation.go]: The same story as in `api_validation.go`, can be refactored.

**Pull request merged**